pkg/download/downloader.go

```diff
@@ -170,44 +60,38 @@ func suspiciousPath(path string) error {
 	return nil
 }
 
-func detectMIMEType(at io.ReaderAt) (string, error) {
-	buf := make([]byte, 512)
-	n, err := at.ReadAt(buf, 0)
-	if err != nil && err != io.EOF {
-		return "", errors.Wrap(err, "failed to read first 512 bytes")
-	}
-	if n < 512 {
-		glog.V(5).Infof("Did only read %d of 512 bytes to determine the file type", n)
-	}
+func isSuspiciousArchive(path string) error {
+	return archiver.Walk(path, func(f archiver.File) error {
+		switch h := f.Header.(type) {
+		case *tar.Header:
+			return suspiciousPath(h.Name)
+		case zip.FileHeader:
+			return suspiciousPath(h.Name)
+		default:
+			return errors.Errorf("Unknow header type: %T", h)
```

Comment:
typo

Reply:
Also: Error strings should not be capitalized
Ref: https://github.com/golang/go/wiki/CodeReviewComments#error-strings